src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Remove.cs

```diff
@@ -0,0 +1,579 @@
+                            {
+                                fValid = false;
+                                errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrRemoveAllArg);
+                            }
+                        }
+
+                        continue;
+                    }
+                    else
+                    {
+                        fValid = false;
+                        errors.EnsureError(args[i], TexlStrings.ErrNeedRecord_Arg, args[i]);
+                        continue;
+                    }
+                }
+
+                var collectionAcceptsRecord = collectionType.Accepts(argType.ToTable(), exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules);
+                var recordAcceptsCollection = argType.ToTable().Accepts(collectionType, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules);
+
+                // The item schema should be compatible with the collection schema.
+                if (!collectionAcceptsRecord && !recordAcceptsCollection)
+                {
+                    fValid = false;
+                    if (!SetErrorForMismatchedColumns(collectionType, argType, args[i], errors, context.Features))
+                    { 
+                        errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrCollectionDoesNotAcceptThisType);
+                    }
+                }
+                
+                // Only warn about no-op record inputs if there are no data sources that would use reference identity for comparison.
+                else if (!collectionType.AssociatedDataSources.Any() && !recordAcceptsCollection)
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Warning, args[i], TexlStrings.ErrCollectionDoesNotAcceptThisType);
+                }
+
+                if (!context.AnalysisMode)
+                {
+                    // ArgType[N] (0<N<argCount) must match all the fields with the data source.
+                    bool checkAggregateNames = argType.CheckAggregateNames(collectionType, args[i], errors, context.Features, SupportsParamCoercion);
+
+                    // The item schema should be compatible with the collection schema.
+                    if (!checkAggregateNames)
+                    {
+                        fValid = false;
+                        if (!SetErrorForMismatchedColumns(collectionType, argType, args[i], errors, context.Features))
+                        {
+                            errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrTableDoesNotAcceptThisType);
+                        }
+                    }
+                }
```

Comment:
This check wasn't in the original Remove implementation; what is the scenario that it is covering? In the existing implementation, there is already a check for the types of the arguments compared to the data source / collection.

Reply:
My understanding is that the returning types are different between PA/PFx.

Reply:
If I understand correctly, this is not about the return type of the function, but for input validation. And as I mentioned in another comment, whenever we expose Power Fx V1 compatibility in Power Apps, it should have the same behavior as the C# interpreter.